aten/src/ATen/native/RNN.cpp

```diff
@@ -68,6 +68,11 @@ using CellParamsSerializationType = std::tuple<
 struct CellParamsBase : torch::CustomClassHolder {
   virtual Tensor matmul_ih(const Tensor& input) const = 0;
   virtual Tensor matmul_hh(const Tensor& h) const = 0;
+  // by default doing nothing. CellParams will override this
+  // to define correct behavior for LSTMs with projections
+  virtual Tensor matmul_hr(const Tensor& h) const {
```

Comment:
The reason is that it's easier to define it here, so that this default behavior can be directly re-used in all function that don't support projections (e.g. quantized cells). I added explanation to this in the code comments as well